lib/index.ts

```diff
@@ -84,6 +84,58 @@ export function concatRegex(...regexes: (string | RegExp)[]) {
   return new RegExp(source);
 }
 
+function _permutations(permutation: (string | RegExp)[]) {
+  const permutations: (string | RegExp)[][] = [];
+
+  function permute(array: (string | RegExp)[], length: number) {
```

Comment:
```suggestion
  // Heap's algorithm
  function permute(array: (string | RegExp)[], length: number) {
```
Just so people know what they're looking at.